Tests/NewRelic/LoggingInteractorDecoratorTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of Ekino New Relic bundle.
+ *
+ * (c) Ekino - Thomas Rabaix <thomas.rabaix@ekino.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ekino\Bundle\NewRelicBundle\Tests\NewRelic;
+
+use Ekino\Bundle\NewRelicBundle\NewRelic\LoggingInteractorDecorator;
+use Ekino\Bundle\NewRelicBundle\NewRelic\NewRelicInteractorInterface;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+
+class LoggingInteractorDecoratorTest extends TestCase
+{
+    /**
+     * @dataProvider provideMethods
+     */
+    public function testGeneric(string $method, array $arguments, $return)
+    {
+        $logger = $this->createMock(LoggerInterface::class);
+        $decorated = $this->createMock(LoggingInteractorDecorator::class);
+        $interactor = new LoggingInteractorDecorator($decorated, $logger);
+
+        $logger->expects($this->once())->method('debug');
+        $decorated->expects($this->once())->method($method)
+            ->with(...$arguments)
+            ->willReturn($return);
+
+        $result = $interactor->$method(...$arguments);
+
+        $this->assertSame($return, $result);
+    }
+
+    public function provideMethods()
+    {
+        $reflection = new \ReflectionClass(NewRelicInteractorInterface::class);
```

Comment:
Have you seen nyholm/nsa?

Reply:
I didn't know that package. The idea here is to retrieve methods with their arguments in order to automate calls, and it looks like this is not the purpose of this package. Did I missed a point?

Reply:
No, I missed the point. (reviewed on my phone earlier) This peace of code you write is pretty cool. I may port it to nyholm/nsa.